File: classlog-frontend/src/app/model/entities/lesson.dto.ts
import {UserDto} from "./user-dto";
import {ClassDto} from "./class-dto";

export interface LessonDto {
  lessonId: number;
  createdByUser: UserDto;
  lessonClass: ClassDto;
  lessonDate?: Date;
  subject?: string;
  content?: string;
}

File: classlog-backend/src/main/java/com/example/classlog/dto/LessonDto.java
package com.example.classlog.dto;

import com.example.classlog.entities.User;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class LessonDto {
    private Long lessonId;
    private UserDto createdByUser;
    private ClassDto lessonClass;
    private LocalDateTime lessonDate;
    private String subject;
    private String content;
}